var/spack/repos/builtin/packages/py-dask-expr/package.py

```diff
@@ -21,5 +21,6 @@ class PyDaskExpr(PythonPackage):
 
     # Can't do circular run-time dependencies yet?
     # depends_on("py-dask@2024.7.1", type="run")
-    depends_on("py-pyarrow@7: +dataset", type="run")
+    depends_on("py-pyarrow@7:", type="run")
+    depends_on("arrow+dataset")
```

Comment:
(Optional) I can't quite confirm how this variant is imposed by `dask-expr`.

Reply:
I can't either.